contracts/Compass.vy

```diff
@@ -164,67 +172,53 @@ def submit_logic_call(consensus: Consensus, args: LogicCallArgs, message_id: uin
     # check if the supplied current validator set matches the saved checkpoint
     assert self.last_checkpoint == self.make_checkpoint(consensus.valset), "Incorrect Checkpoint"
     # signing data is keccak256 hash of abi_encoded logic_call(args, message_id, compass_id, deadline)
-    args_hash: bytes32 = keccak256(_abi_encode(args, message_id, COMPASS_ID, deadline, method_id=method_id("logic_call((address,bytes),uint256,bytes32,uint256)")))
+    args_hash: bytes32 = keccak256(_abi_encode(args, message_id, compass_id, deadline, method_id=method_id("logic_call((address,bytes),uint256,bytes32,uint256)")))
     # check if enough validators signed args_hash
     self.check_validator_signatures(consensus, args_hash)
     # make call to logic contract
     raw_call(args.logic_contract_address, args.payload)
-    log LogicCallEvent(args.logic_contract_address, args.payload, message_id)
-
-@internal
-def _safe_transfer_from(_token: address, _from: address, _to: address, _value: uint256):
-    _response: Bytes[32] = raw_call(
-        _token,
-        _abi_encode(
-            _from, _to, _value,
-            method_id=method_id("transferFrom(address,address,uint256)")),
-        max_outsize=32
-    )  # dev: failed transferFrom
-    if len(_response) > 0:
-        assert convert(_response, bool), "TransferFrom failed"
+    event_id: uint256 = unsafe_add(self.last_event_id, 1)
+    self.last_event_id = event_id
+    log LogicCallEvent(args.logic_contract_address, args.payload, message_id, event_id)
 
 @external
 def send_token_to_paloma(token: address, receiver: String[64], amount: uint256):
     _balance: uint256 = ERC20(token).balanceOf(self)
-    self._safe_transfer_from(token, msg.sender, self, amount)
+    assert ERC20(token).transferFrom(msg.sender, self, amount, default_return_value=True), "TF fail"
     _balance = ERC20(token).balanceOf(self) - _balance
     assert _balance > 0, "Zero Transfer"
-    log SendToPalomaEvent(token, msg.sender, receiver, amount)
-
-@internal
-def _safe_transfer(_token: address, _to: address, _value: uint256):
-    _response: Bytes[32] = raw_call(
-        _token,
-        _abi_encode(
-            _to, _value,
-            method_id=method_id("transfer(address,uint256)")),
-        max_outsize=32
-    )  # dev: failed transferFrom
-    if len(_response) > 0:
-        assert convert(_response, bool), "TransferFrom failed"
+    event_id: uint256 = unsafe_add(self.last_event_id, 1)
+    self.last_event_id = event_id
+    log SendToPalomaEvent(token, msg.sender, receiver, amount, event_id)
 
 @external
-def submit_batch(consensus: Consensus, token: address, args: TokenSendArgs, message_id: uint256, deadline: uint256):
+def submit_batch(consensus: Consensus, token: address, args: TokenSendArgs, message_id: uint256, batch_id: uint256, deadline: uint256):
```

Comment:
You can remove `message_id` from this one (or simply rename it `batch_id`, as that is what we're sending here already).

I don't think we can send you the same `message_id` as we do in `SubmitLogicCall`. A logic call and a gravity batch are two completely different entities in Paloma, so there is no common ID that's increased. We'll keep the `message_id` for the logic calls, but we can remove them from the `SubmitBatch` and use `batch_id` instead.